src/main/java/org/mule/runtime/http/policy/api/HeadersAwarePointcut.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.http.policy.api;
+
+import org.mule.runtime.policy.api.PolicyPointcut;
+
+/**
+ * Marker interface for classes that implement {@link PolicyPointcut}.
+ * <p>
+ * If the class that extends {@link HttpPolicyPointcutParameters} also implements this interface, the headers will be taken into
+ * account when evaluating the pointcut. Otherwise, they will be ignored.
+ *
+ * @since 4.3
+ */
+public interface HeadersAwarePointcut extends PolicyPointcut {
```

Comment:
Shouldn't there be also a `ResourceAwarePointcut` interface to avoid cache misses for different paths/methods when resource level pointcuts are not used?

Reply:
will add that, yes